monailabel/transform/writer.py

```diff
@@ -104,15 +185,28 @@ def __call__(self, data):
             output_file = tempfile.NamedTemporaryFile(suffix=ext).name
             logger.debug(f"Saving Image to: {output_file}")
 
+            if self.is_multichannel_image(image_np):
+                if ext != ".seg.nrrd":
+                    logger.debug(
+                        f"Using extension '{ext}' with multi-channel 4D label will probably fail"
+                        + "Consider to use extension '.seg.nrrd'"
+                    )
+                labels = data.get("labels")
+                color_map = data.get("color_map")
+                logger.debug("Using write_seg_nrrd...")
+                write_seg_nrrd(image_np, output_file, dtype, affine, labels, color_map)
             # Issue with slicer:: https://discourse.itk.org/t/saving-non-orthogonal-volume-in-nifti-format/2760/22
-            if self.nibabel and ext.lower() in [".nii", ".nii.gz"]:
+            elif self.nibabel and ext.lower() in [".nii", ".nii.gz"]:
                 logger.debug("Using MONAI write_nifti...")
                 write_nifti(image_np, output_file, affine=affine, output_dtype=dtype)
             else:
                 write_itk(image_np, output_file, affine, dtype, compress)
 
         return output_file, output_json
 
+    def is_multichannel_image(self, image_np):
+        return len(image_np.shape) == 4 and image_np.shape[-1] > 1
```

Comment:
channel last?  normally all images/labels/predictions are processed as channel first in post transform... we better cross check by running some examples e2e.. 

@diazandr3s  can help to verify this feature...

Reply:
Happy to help :)
Should we wait until this PR (https://github.com/Project-MONAI/MONAI/pull/4259) is merged?

Reply:
you can checkout the remote branch and verify..

Reply:
https://github.com/Project-MONAI/MONAI/pull/4259 is merged now. I merged upstream/main into this branch

Reply:
Indeed this seemed wrong, I updated the code to check the first dimension instead.

FYI:

**[pynrrd](https://pypi.org/project/pynrrd/)** choses the following convention
```
default index_order='F'
nrrd.writer.write(data)
index_order : {'C', 'F'}, optional
    Specifies the index order used for writing. Either 
    'C' (C-order) 
    	where the dimensions are ordered from
    	slowest-varying to fastest-varying (e.g. (z, y, x)), 
    'F' (Fortran-order)
    	where the dimensions are ordered
    	from fastest-varying to slowest-varying (e.g. (x, y, z)).
```
We chose 'C' as a default here

**torch** has the following convention: `NxCxHxW`

At the moment, in our monailabel-app, we use (5, 4280, 3520, 1) = (channels, width, height, batch)